File: internal/model/permission.go
package model

type Permission struct {
	ID             ID            `gorm:"primarykey" json:"id"`
	ParentID       *ID           `gorm:"index" json:"parentId"`
	Name           string        `gorm:"index" json:"name"`
	Path           string        `gorm:"index" json:"path"`
	Icon           string        `gorm:"index" json:"icon"`
	PermissionCode string        `gorm:"index" json:"permissionCode"`
	Sort           *int          `gorm:"index" json:"sort"`
	Visible        *bool         `gorm:"index;default:true" json:"visible"`
	Type           *int          `gorm:"index" json:"type"`
	PermissionUrls string        `gorm:"column:permission_urls" json:"permissionUrls"` // 多个Url用英文逗号分割
	Children       []*Permission `gorm:"-" json:"children,omitempty"`
}

func (p *Permission) TableName() string {
	return "permission"
}

File: frontend/src/api/permission.ts
import request from '@/utils/request'

export interface Permission {
  id: string | number
  parentId?: string | number
  name: string
  permissionCode: string // This maps to permissionCode from backend
  type: number // 1: Menu, 2: Button
  path?: string
  icon?: string
  visible?: number | boolean
  sort?: number
  children?: Permission[]
}

export function getPermissionTree(params?: any) {
  return request<Permission[]>({
    url: '/common/user/permissions',
    method: 'get',
    params
  })
}
